compiler/rustc_driver_impl/src/lib.rs

```diff
@@ -175,6 +177,7 @@ pub trait Callbacks {
     }
     /// Called after analysis. Return value instructs the compiler whether to
     /// continue the compilation afterwards (defaults to `Compilation::Continue`)
+    /// At this point, the MIR (Middle-Level Intermediate Representation) becomes available.
```

Comment:
I don't know if it's productive to say "becomes available". The way that the query system works means that these IRs are available *whenever* you call it.

These callbacks are useful to know that a certain phase of compilation has been performed. Like `after_expansion` ensures that macros have been expanded and any errors have been reported from that -- HIR is actually not created until during analysis. Similarly with `after_analysis`, you can ensure that any type-checking based errors have been emitted.